src/Nest/CommonOptions/TimeUnit/Time.cs

```diff
@@ -20,7 +21,7 @@ public class Time : IComparable<Time>, IEquatable<Time>
 		private const double MicrosecondsInAMillisecond = 10;
 
 		private static readonly Regex ExpressionRegex =
-			new Regex(@"^(?<factor>[-+]?\d+(?:\.\d+)?)\s*(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros))?$",
+			new Regex(@"^(?<factor>[-+]?[\d\.e\-]+?)\s*(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros))?$",
```

Comment:
Going over it again i think your regex can be simplified like this;

`^(?<factor>[+\-]?(?<numeric>(?:\d+(?:\.\d*)?)|(?:\.\d+))(?:[e](?<exponent>[+\-]?\d+))?)\s{0,10}(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros))?$`

`<numeric>` is no longer optional so the if pattern on exponent is no longer needed and the negative lookahead from my previous comment can go as well.